lib/components/form/submission/list.vue

```diff
@@ -209,8 +208,15 @@ export default {
         this.chunks = 1;
       } else {
         if (data.value != null) {
+          const lastSubmission = this.submissions[this.submissions.length - 1];
+          const lastSubmissionDate = lastSubmission.__system.submissionDate;
           for (const submission of data.value) {
-            if (!this.instanceIds.has(submission.__id)) {
+            // If a submission has been created since the initial fetch or last
+            // refresh, the latest chunk of submissions may include a newly
+            // created submission or a submission that is already shown in the
+            // table.
+            if (submission.__system.submissionDate <= lastSubmissionDate &&
```

Comment:
hm. is this overzealous? does the `.has` check not do it?

Reply:
I think the `.has` check might not catch all submissions created since the submission shown in the first row was created, since such submissions wouldn't have been seen before, so wouldn't be caught by `.has`. I think that would be pretty rare though, because a whole chunk's worth of new submissions would have to be created in-between two chunks. But if that did happen, I think something like this would be possible:

* The user navigates to the submissions page and loads the first 250 submissions, then steps out for lunch.
* During lunch, 300 new submissions are created.
* After lunch, the user fetches the second chunk of submissions (`$top=250, $skip=250`). That returns the last 50 of the 300 newly created submissions, as well as 200 submissions that are already shown in the table.
* The 50 submissions would pass the `.has` test, but we wouldn't want to include them in the table.

That is definitely pretty unlikely. But the way I'm thinking about it, the `submissionDate` check helps ensure that there are no out-of-order submissions, then the `.has` check ensures that there are no duplicates.